urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/HTTPHandler.java

```diff
@@ -0,0 +1,1131 @@
+/**
+ *   ikrs.httpd - A free java http server based on ikrs.yucca.
+ *   Copyright (C) 2012 Ikaros Kappler
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+
+package ikrs.httpd;
+
+/**
+ * This is the main handler class that will be bound as a listener to the yucca server.
+ *
+ * @author Ikaros Kappler
+ * @date 2012-05-15
+ * @modified 2013-04-17 Ikaros Kappler (shared handler instance added).
+ * @version 1.0.0
+ **/
+
+
+import java.io.BufferedReader;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.DatagramSocket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.Socket;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+
+
+
+import ikrs.httpd.resource.FileSystemResourceAccessor;
+import ikrs.httpd.resource.DefaultDirectoryResource;
+import ikrs.yuccasrv.ConnectionUserID;
+import ikrs.yuccasrv.TCPAdapter;
+import ikrs.yuccasrv.socketmngr.BindManager;
+
+import ikrs.typesystem.*;
+
+import ikrs.util.CaseInsensitiveComparator;
+import ikrs.util.CustomLogger;
+import ikrs.util.DefaultCustomLogger;
+import ikrs.util.DefaultEnvironment;
+import ikrs.util.Environment;
+import ikrs.util.EnvironmentFactory;
+import ikrs.util.FileExtensionKeyMap;
+import ikrs.util.HexDumpOutputStream;
+import ikrs.util.KeyValueStringPair;
+import ikrs.util.MapFactory;
+import ikrs.util.TreeMapFactory;
+
+import ikrs.util.session.*;
+
+
+public class HTTPHandler 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 70 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.